apps/prairielearn/src/lib/groups.sql

```diff
@@ -9,11 +9,41 @@ WHERE
 
 -- BLOCK create_group
 WITH
+  next_group_number AS (
+    SELECT
+      SUBSTRING(
+        g.name
+        FROM
+          6
+      )::INTEGER + 1 AS group_number
+    FROM
+      groups AS g
+      JOIN group_configs AS gc ON (gc.id = g.group_config_id)
+    WHERE
+      gc.assessment_id = $assessment_id
+      AND g.name ~ '^group[0-9]+$'
+      AND g.deleted_at IS NULL
+    ORDER BY
+      group_number DESC
+    LIMIT
+      1
+  ),
```

Comment:
Aside: while it's almost definitely outside the scope of this PR, I'd thought it would be cute so use something like https://www.npmjs.com/package/unique-names-generator to generate unique group names.